internal/domain/tenant/repository.go

```diff
@@ -0,0 +1,10 @@
+package tenant
+
+import (
+	"context"
+)
+
+type Repository interface {
+	Create(ctx context.Context, tenant *Tenant) error
+	GetById(ctx context.Context, id string) (*Tenant, error)
```

Comment:
```suggestion
	GetByID(ctx context.Context, id string) (*Tenant, error)
```

We are using `ID` for `id` fields everywhere so lets make it consistent